### ISSUE TITLE
DeprecationWarning: Flags not at the start of the expression

### ISSUE THREAD

**Issue by burhanteo** [bottlepy/bottle#949]
```
from bottle import route, run, template

@route('/hello/<name>')
def index(name):
    return template('<b>Hello {{name}}</b>!', name=name)

run(host='localhost', port=8080, debug=True, reloader=True)
```
When I run this code($ python3 hello.py) I get that error:

```
Bottle v0.13-dev server starting up (using WSGIRefServer())...
Listening on http://localhost:8080/
Hit Ctrl-C to quit.

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/sre_parse.py:864: DeprecationWarning: Flags not at the start of the expression \{\{((?:(?mx)(       (truncated)
  p = _parse_sub(source, pattern, True, False)
```
Bottle is awesome. Thanks for your great work!

**Comment by defnull** [comment]
Thats this one: https://github.com/bottlepy/bottle/blob/master/bottle.py#L4007
It re-uses a pattern with flags in it.

**Comment by bulletmark** [comment]
I get this warning printed twice on app startup using Arch Linux with python 3.6. Slightly different to OP report.
```
/usr/lib/python3.6/re.py:301: DeprecationWarning: Flags not at the start of the expression ((?m)[urbURB]?(?:''( (truncated)
  p = sre_compile.compile(pattern, flags)
/usr/lib/python3.6/sre_parse.py:763: DeprecationWarning: Flags not at the start of the expression \{\{((?:((?m)[urbURB (truncated)
  p = _parse_sub(source, state, sub_verbose)
```

**Comment by admin-zou** [comment]
mybe your method is error.
in your fuction you use GET，but the web use POST method to get data,
so you need add " response.headers['Access-Control-Allow-Origin'] = '*' "  

for exmaple：

@app.route('/hello',method='POST')
def hello():
    response.headers['Access-Control-Allow-Origin'] = '*'
    return 'hello world'

**Comment by xeno1701** [comment]
I also get this on windows:
`C:\WP36\python-3.6.2rc2.amd64\lib\site-packages\bottle-0.13.dev0-py3.6.egg\bottle.py:4082: DeprecationWarning: Flags not at the start of the expression '\\{\\{((?:(?mx)(      ' (truncated)
  patterns = [re.compile(p % pattern_vars) for p in patterns]`

**Comment by floggle** [comment]
I'm getting this too.
Python 3.6.3, bottle 0.12.13.

This is the Python Change that you're going to need to work around to remove the Deprecation message:
https://github.com/python/cpython/commit/db99e6ee57c5070e6ef82fd2eaa528f09f3948dc

It's triggered by this line in bottle.py:
            patterns = [re.compile(p%pattern_vars) for p in patterns]


These are my declarations:
@web_server.route('/')
def index_page():
	return template('index.html', root='')

@web_server.route('/delete_url')
def delete_url():
	return template('delete_url.html', root='')

**Comment by joshualevy2** [comment]
This is hitting me as well.  Any plans to fix?  It clouds my automated test suite results.
Python is 3.6.4.  Bottle is 0.12.13.

**Comment by acidsploit** [comment]
https://github.com/bottlepy/bottle/issues/930

**Comment by benpdavison** [comment]
Hello All, just wanted to say, I only got this issue when accessing parsed json, and only when accessing like so:
```
data = request.json
x = request['x']
```

If i use double-quoutes it is fine:
```
data = request.json
x = request["x"]
```

**Comment by wendyyu21** [comment]
have the same problem as mentioned, so how to fix it?

**Comment by Valmadson** [comment]
in Postman and firefox do not return error, call in Chrome return the message.

**Comment by olivierpons** [comment]
Python3.7 under Windows: same warning:

```
127.0.0.1 - - [11/Nov/2018 22:28:48] "GET /level/aruba2 HTTP/1.1" 200 2514
C:\Python37\lib\site-packages\bottle.py:3468: DeprecationWarning: Flags not at the start of the expression "((?m)[urbURB]?(?:''(" (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
C:\Python37\lib\site-packages\bottle.py:3468: DeprecationWarning: Flags not at the start of the expression '\\{\\{((?:((?m)[urbURB' (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
```

**Comment by dmahugh** [comment]
+1 I'm running Bottle 0.12.13 on Python 3.7.0, and get this error twice at startup:

```
bottle.py:3468: DeprecationWarning: Flags not at the start of the expression "((?m)[urbURB]?(?:''(" (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
```

**Comment by Dszmid91** [comment]
+1 same error 

bottle.py:3468: DeprecationWarning: Flags not at the start of the expression "((?m)[urbURB]?(?:''(" (truncated)
patterns = [re.compile(p%pattern_vars) for p in patterns]

**Comment by SMbulius** [comment]
+1 some erro for my ('Linux Mint 19 Cinamon', Bottle v0.12.13, Python v3.6.7)

```
/usr/lib/python3/dist-packages/bottle.py:3468: DeprecationWarning: Flags not at the start of the expression "((?m)[urbURB]?(?:''(" (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
/usr/lib/python3/dist-packages/bottle.py:3468: DeprecationWarning: Flags not at the start of the expression '\\{\\{((?:((?m)[urbURB' (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
```

I get error when call in Chrome (v70.0.3538.110) and Opera (v57.0). No error => Firefox (v63.03).

**Comment by defnull** [comment]
Backport of the fix is now in 0.12.15
